src/embedding-model/Dockerfile

```
@@ -0,0 +1,39 @@
+FROM python:3.11-slim-buster
+
+# Environment variabless
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYENV_SHELL=/bin/bash
+ENV LANG=C.UTF-8
+ENV PYTHONBUFFERED=1
+
+# Install system dependencies and pipenv
+RUN set -ex; \
+  apt-get update && \
+  apt-get upgrade -y && \
+  apt-get install -y build-essential curl && \
```

```suggestion
  apt-get install -y build-essential curl libgomp1 && \
```

You can try this to see if it fixes the Docker issue.

---

Unfortunately this doesn't. The bug is related to https://bugzilla.redhat.com/show_bug.cgi?id=1722181 and we are having `glibc 2.28` with `buster` so the bug is exposed. Workaround is to set `LD_PRELOAD` but the preload path contains a hash that is hard to know in advance. Switching to a newer base image of `bullseye` would give higher versions of `glibc` that may fix the issue.